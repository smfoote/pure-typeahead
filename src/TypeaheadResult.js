import React, {Component} from 'react'

export default class TypeaheadResult extends Component {
  render() {
    const { children } = this.props;
    return (
      <typeahead-result
        onClick={this.props.onSelect}
        class={this.props.isHighlighted ? 'highlighted' : null}
      >
        {children}
      </typeahead-result>
    );
  }
}
