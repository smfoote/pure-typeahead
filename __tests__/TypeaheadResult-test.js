import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import TypeaheadResult from '../src/TypeaheadResult';

Enzyme.configure({ adapter: new Adapter() });

const shallowSetup = () => {
  const props = {
    onSelect: jest.fn(),
    isHighlighted: false,
  };
  return {
    wrapper: shallow(<TypeaheadResult {...props}>Las Vegas</TypeaheadResult>),
    props
  };
};

describe('Typeahead', () => {
  it('renders self and subcomponents', () => {
    const { wrapper } = shallowSetup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders self and subcomponents, isHighlighted is true', () => {
    const { props } = shallowSetup();
    props.isHighlighted = true;
    const wrapper = shallow(<TypeaheadResult {...props}>Las Vegas</TypeaheadResult>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call onSelect when clicked', () => {
    const { props, wrapper } = shallowSetup();
    wrapper.find('typeahead-result').simulate('click');
    expect(props.onSelect).toHaveBeenCalled();
  });
});
