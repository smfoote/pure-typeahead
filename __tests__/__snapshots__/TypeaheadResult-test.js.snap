// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeaheadResult renders self and subcomponents 1`] = `
<typeahead-result
  class={null}
  onClick={[Function]}
>
  Las Vegas
</typeahead-result>
`;

exports[`TypeaheadResult renders self and subcomponents, isHighlighted is true 1`] = `
<typeahead-result
  class="highlighted"
  onClick={[Function]}
>
  Las Vegas
</typeahead-result>
`;
